.github/workflows/release-console.yml

```diff
@@ -0,0 +1,88 @@
+name: Release Console
+
+on:
+  workflow_dispatch:
+    inputs:
+      releaseVersion:
+        description: 'Release version (e.g., v1.2.3)'
+        required: true
+      commitHash:
+        description: 'Commit hash to tag'
+        required: true
+      environment:
+        description: 'Deployment environment (main or testnet)'
+        required: true
+        default: 'main'
+
+env:
+  E2E_TEST_WORKFLOW: ./.github/workflows/ci-trading-python-tests.yml
+
+jobs:
+  setup_release_branch:
+    runs-on: ubuntu-latest
+    outputs:
+      branch_name: ${{ steps.prepare_branch.outputs.branch_name }}
+    steps:
+      - name: Checkout code at commit hash
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.inputs.commitHash }}
+
+      - name: Create Tag
+        run: git tag ${{ github.event.inputs.releaseVersion }} ${{ github.event.inputs.commitHash }}
+
+      - name: Push Tag
+        run: git push origin ${{ github.event.inputs.releaseVersion }}
+
+      - name: Prepare release branch
+        id: prepare_branch
+        run: |
+          BRANCH_NAME="release/${{ github.event.inputs.environment }}/${{ github.event.inputs.releaseVersion }}"
+          echo "::set-output name=branch_name::$BRANCH_NAME"
+          git fetch --all
+          git checkout -b $BRANCH_NAME origin/${{ github.event.inputs.environment }}
+          git reset --hard ${{ github.event.inputs.commitHash }}
+          git push --force --set-upstream origin $BRANCH_NAME
+        if: ${{ github.event.inputs.environment == 'testnet' }}
+
+  run_e2e_tests:
+    needs: setup_release_branch
+    uses: ${{ env.E2E_TEST_WORKFLOW }}
+    with:
+      github-sha: ${{ needs.setup_release_branch.outputs.branch_name }}
+
+  await_approval:
+    needs: run_e2e_tests
+    runs-on: ubuntu-latest
+    if: ${{ always() && github.event.inputs.environment == 'testnet' }}
```

Comment:
Based on your goal, I understood you want to require an approval only when the e2e tests fail. As a result, I think `always()` is not the right condition. It should be condition based on `run_e2e_tests`'s failure . 

I am not sure that's possible (or easy) to do with the current splitting of the jobs. I think you would be better off with the manual approval to be a step inside the job `run_e2e_tests`, that would only be triggered if the step running the e2e tests fails.

```yaml
run_e2e_tests:
  steps:
    - name: Run e2e tests
      id: e2e-tests
      run: ...
    - name: Requires manual approval
      if: steps.e2e-tests.outcome != 'success'
      run: ...
```

What do you think ?

Reply:
I have updated now to always require the approval, that way we can choose to merge before the end of the test run etc. It gives us more control.